### DISCUSSION TITLE
Add the ability to customize the base of author URLs

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Authors#1220] [Feature Requests]
At the moment, this is all that people see:
<img width="804" alt="Screen Shot 2020-12-23 at 11 04 13 AM" src="https://user-images.githubusercontent.com/3868028/103015582-b268bf80-450e-11eb-8527-b4dd09b4c831.png">

@heyjoecampbell mentions:

> Add a parameter to configure the author permalink base:
> 
> https://wordpress.org/plugins/edit-author-slug/
> https://www.wpbeginner.com/plugins/how-to-change-author-url-slug-and-base-in-wordpress/
> 
> This is how I accomplish it:
> https://www.isitwp.com/change-the-author-slug-url-base/

**Comment by andergmartins** [comment]
This conflicts with #288, where we added a task for syncing the author's URL and the user's nickname, a requirement for Pemissions integration. If we implement this option we need to fix the integration with Permissions.

**Comment by andergmartins** [comment]
Maybe we can implement this by changing the user's nickname column and syncing the author term's slug. This way we keep it compatible with the Permissions plugin. Users are trying to achieve this using the plugin Edit Author Slug.

**Comment by stevejburge** [comment]
One more request for this today.

They want to change the base slug from "author" to "contributor"

**Comment by heyjoecampbell** [comment]
Contributor is a more accurate and inclusive term - ultimately this needs changed in the WP core

**Comment by stevejburge** [comment]
Thanks @heyjoecampbell 

At the moment, it looks like changing the base is possible with https://wordpress.org/plugins/edit-author-slug/

**Comment by heyjoecampbell** [comment]
Thanks @stevejburge 

I prefer using Code Snippets when possible:

/*
Change Author Base Slug
source: https://www.isitwp.com/change-the-author-slug-url-base/
*/

add_action('init', 'cng_author_base');
function cng_author_base() {
    global $wp_rewrite;
    $author_slug = 'contributors'; // change slug name
    $wp_rewrite->author_base = $author_slug;
}
